레스토랑 조회 구현하기

hanvit7 opened the pull request:
안녕하세요 트레이너님 리뷰요청드립니다.

이전 보다 덜 완벽하게, 우선 의식의 흐름대로? 진행하고 있습니다.
잘되고 있는것인지 모르겠지만 이전 보다는 그래도 좀 속도가 나는 것 같습니다. 
비동기까지는 아직 갈길이 머네요 ㅎㅎ

hanvit7 committed:
테스트 - App에 region 이름 테스트 추가

hanvit7 committed:
구현 - App에 region 이름 구현

hanvit7 committed:
테스트 - RegionContainer 추가

- 지역명으로 된 리스트 테스트 확인

hanvit7 committed:
구현 - RigionContainer 생성

hanvit7 committed:
테스트 - Rigion 추가

- 지역명으로 된 리스트 테스트 확인

hanvit7 committed:
구현 - Region 구현

- 지역명으로 된 리스트 테스트 확인

hanvit7 committed:
구현 - RigionContainer에 redux 추가

- useSelector로 regions 상태 조회

hanvit7 committed:
구현 - redux boilerplate 추가

hanvit7 committed:
테스트 - RegionContainer 테스트 추가

- redux에서 regions 상태 조회 테스트 추가

hanvit7 committed:
테스트 - App에 redux mock 추가

hanvit7 committed:
테스트 - RegionContainer 테스트 regions mock 수정

- region : { id, name} 구조로 변경

hanvit7 committed:
리팩토링 - regions 더비 데이터 추출

- RegionContainer.test 리팩토링, 오류 수정
- reducer 초기값에 더미 데이터 적용

류

hanvit7 committed:
구현 - Region 생성

hanvit7 committed:
테스트 - 오류 수정

app
- 상태값 추가

Region
- 리스트 갯수 테스트 수정
- 지역명 테스트 수정

RegionContainer
- 리스트 확인 테스트 수정

hanvit7 committed:
테스트 - Button 테스트 추가

-  이름과 버튼 확인

hanvit7 committed:
구현 - Button 구현, Region에 적용

Kihyun92 commented on src/Region.jsx:
```diff
@@ -0,0 +1,13 @@
+import Button from './Button';
+
+export default function Rigion({ regions }) {
```
`Rigion` 으로 된 부분이 있는 것 같아요! 전반적으로 확인해주세요~

Kihyun92 commented on src/Region.jsx:
```diff
@@ -0,0 +1,13 @@
+      {regions.map((region) => (
+        <li key={region.id}>
+          <Button name={region.name} />
+        </li>
```
구조분해할당을 사용해도 좋을 것 같군요~

Kihyun92 commented on src/Button.jsx:
```diff
@@ -0,0 +1,7 @@
+export default function Button({ name }) {
```
공통 컴포넌트화 하기에는 Button 컴포넌트의 역할이 조금 단순한 것 같습니다 😅  

만약 반복되는 로직이 들어가거나 동일한 스타일의 버튼이 반복되는 경우가 아니라면 굳이 버튼 컴포넌트를 벌써 공통화 해줄 필요는 없을 것 같아요 ㅎㅎ

Kihyun92 commented on src/Region.test.jsx:
```diff
@@ -0,0 +1,12 @@
+  const { getByText, getAllByRole } = render(<Region regions={regions} />);
+
+  expect(getAllByRole('listitem')).toHaveLength(regions.length);
+  expect(getByText(regions[1].name)).not.toBeNull();
```
만약 fixtures가 변경되어서 regions 배열에 인자가 한개가 된다면 어떻게 될까요?

테스트에서 저런 변경에 신경쓰지 않고 확인할 수 있도록 처리해줄 수 없을까요?

Kihyun92 reviewed (changes requested):
커밋을 보니 TDD cycle을 잘 지켜주고 계신 것 같아요 👍🏼 

테스트 코드도 리팩터링 대상입니다~ 리팩터링 단계에서 불필요한 반복이나 코드 개선을 해주면 좋을 것 같아요 ㅎㅎ

Kihyun92 commented:
과제가 어떻게 진행중이신지 궁금해요! 진행에 어려움이 있거나 의논이 필요하면 꼭 말씀해주세요😊

올라온 과제 풀이 영상을 보시고 직접 과제를 해보셔도 좋을 것 같습니다~

hanvit7 committed:
구현 - 지역, 분류 목록 얻기 기본틀 구현

- app에서 loadInitialData 로드 구현
- action에서 loadInitialData 및 setRegions 구현

hanvit7 committed:
테스트 - setRegions 테스트 추가

hanvit7 committed:
구현 - reduce에 setRegions 구현

hanvit7 committed:
테스트 - app에 dispatch 호출 테스트 추가

hanvit7 committed:
구현 - api fetch 구현

- api에서 fetchRegions, fetchRegions 구현 및 mock 추가
- actions에서 fetchRegions, fetchRegions 로드 추가

hanvit7 committed:
테스트 - setCategories 테스트 추가

hanvit7 committed:
구현 - reduce에 setCategories 구현

hanvit7 committed:
리팩토링 - reducer 리팩토링

hanvit7 committed:
구현 - defaultReducer 구현

- TypeError 오류를 없애기 위해 defaultReducer 추가

hanvit7 committed:
구현 - store에 redux-thunk 추가

hanvit7 committed:
오타 수정

hanvit7 committed:
리팩토링 - Region 구조분해할당 적용

hanvit7 committed:
리팩토링 - 불필요한 button 컴포넌트 삭제

hanvit7 committed:
리팩토링 - Region 테스트 수정

- fixtures의 객체 수정과 유연하게 수정

hanvit7 commented:
안녕하세요 트레이너님 
수요일부터 회사일로 바뻐 정신을 못 차리고 있었습니다.
그 여파가 남이있는지 오늘은  몸이 안좋아 하루종일 기절하고 이제 좀 정신차리고 과제 진행중입니다...ㅜㅠ
못다한 부분도 직접 해보고 리뷰 요청 드리겠습니다.

hanvit7 committed:
구현 - region에 버튼과 onClick 이벤트 추가

hanvit7 committed:
테스트 - RegionContainer에 버튼 클릭 이벤트 테스트 추가

hanvit7 committed:
테스트 - reducer에 selectRegion 테스트 추가

hanvit7 committed:
구현 - selectRegion 구현

- actions에 selectRegion 구현
- reducer에 selectRegion 구현
- RegionContainer에 dispatch와 handleClick 구현

hanvit7 committed:
구현 - 버튼에 selectedRegion 표시 구현

hanvit7 committed:
구현 - CategoriesContainer 추가

- 기존 Regions를 이용해서 기본틀 구현

hanvit7 committed:
테스트 - CategoriesContainer 버튼 클릭 테스트 추가

hanvit7 committed:
테스트 - selectCategory 테스트 추가

hanvit7 committed:
구현 - selectCategory 구현

- actions에 selectCategory 구현
- reducer에 selectCategory 구현

hanvit7 committed:
리팩토링 - utils.js에 equal, get 함수 리팩토링

- equal : 오브젝트의 키 값과 value값을 비환
- get : 오브텍트의 키 값의 value값을 반환

hanvit7 committed:
테스트 - utils 테스트 추가

hanvit7 committed:
리팩토링 - utils 적용

hanvit7 committed:
구현 - RestaurantsContainer 추가

- 기존 CategoriesContainer를 이용해서 기본틀 구현

hanvit7 committed:
구현 - loadRestaurants 구현

- CategoriesContaier에서 버튼 클릭시 loadRestaurants 호출
- actions에 loadRestaurants, setRestaurants 구현
- api에 fetchRestaurants 구현

hanvit7 committed:
테스트 - setRestaurants 테스트 추가

hanvit7 committed:
구현 - setRestaurants 구현

Kihyun92 reviewed (commented):
바쁜 한주를 보내신 것 같아요. 이번주도 고생하셨습니다~😊 

커밋을 보니 최대한 TDD cycle을 지키려고 노력하신 것 같아 좋습니다 👍🏼 

과제 풀이 영상을 꼭 보시고, 아샬님은 어떻게 접근하시는지 봐보시면 좋을 것 같습니다 ㅎㅎ

슬랙 채널에 회고 남겨주세요~

hanvit7 committed:
구현 - loadRestaurants 구현

- actions : loadRestaurants 구현
- api : fetchRestaurants 쿼리문 수정
- app : RestaurantsContainer 추가
- RegionContainer : loadRestaurants 추가

hanvit7 committed:
불필요 코드 삭제
